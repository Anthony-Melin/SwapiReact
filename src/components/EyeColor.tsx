import { PropsWithChildren } from 'react'

type EyeColorProps = PropsWithChildren<{ value: string }>

const EyeColor = ({ value, ...restProps }: EyeColorProps) => {
    return (
        <p datatest-id="eyecolor" {...restProps}>
            EyeColor: {value}
        </p>
    )
}

export default EyeColor
