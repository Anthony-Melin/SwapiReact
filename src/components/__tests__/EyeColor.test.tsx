import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import EyeColor from '../EyeColor.tsx'

describe('EyeColor', () => {
    beforeEach(() => {
        render(<EyeColor value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('eyecolor')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('eyecolor')).toHaveTextContent(
            'Eye color: VALUE'
        )
    })
})
