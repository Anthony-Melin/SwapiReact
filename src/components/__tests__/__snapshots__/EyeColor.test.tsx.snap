// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EyeColor > should match snapshot 1`] = `
<p
  data-testid="eyecolor"
>
  Eye color: 
  VALUE
</p>
`;
